pkg/collector/horizontalpodautoscaler.go

```diff
@@ -72,6 +72,32 @@ func HorizontalPodAutoscaler(cfg config.Config, logger logr.Logger, otelcol v1al
 		}
 		metrics = append(metrics, targetCPUUtilization)
 
+		// check if fields are provided by .Spec.Autoscaler
+		// otherwise check deprecated fields
+		maxReplicas := new(int32)
```

Comment:
Could we use defaulting webhook to set the new fields to avoid this logic in this file? 

Also how will the upgrade work?

Reply:
Thanks for the review @pavolloffay! I updated with your suggestion to use the defaulting webhook.

I am generally unfamiliar with the upgrade process, but took a look at [v0_56_0.go](https://github.com/open-telemetry/opentelemetry-operator/blob/main/pkg/collector/upgrade/v0_56_0.go) which upgraded HPA to change the scaling target reference. This upgrade makes sense because the underlying HPA should be altered from scaling on `Deployment` to scaling on `OpenTelemetryCollector`. 

For this PR, the deprecated fields are still supported, so I think underlying HPA will remain unchanged. I don't think it's necessary to upgrade the HPA in this case (unless I'm missing something)